src/sparseml/pytorch/sparsification/pruning/modifier_pruning_layer.py

```diff
@@ -70,6 +70,7 @@ def __init__(
             start_epoch=start_epoch,
             end_epoch=end_epoch,
             update_frequency=-1.0,
+            end_comparator=-1,
```

Comment:
noting that we should not merge this PR yet. this fix uncovered potential issues with the logic surrounding the implementation of `end_comparator=-1`